src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStoreConfig.java

```diff
@@ -128,4 +138,32 @@ public ListBoxModel doFillRegionItems() {
         }
         return regions;
     }
+
+    public FormValidation doCheckContainer(@QueryParameter String container){
+        FormValidation ret = FormValidation.ok();
+        if (StringUtils.isBlank(container)){
+            ret = FormValidation.error("The container name cannot be empty");
+        } else if (!bucketPattern.matcher(container).matches()){
+            ret = FormValidation.error("The container name does not match with S3 bucket rules");
+        }
+        return ret;
+    }
+
+    public FormValidation doCheckPrefix(@QueryParameter String prefix){
+        FormValidation ret = FormValidation.ok();
+        if (StringUtils.isBlank(prefix)){
+            ret = FormValidation.error("Prefix should have a value");
```

Comment:
I do not think this is actually required—AFAIK a blank prefix is fine, and probably the normal case. Just means you have no particular need to use a “subdirectory” of the bucket.